docs/user-guide/advanced_topics/encrypted_communication.md

```diff
@@ -0,0 +1,327 @@
+
+### Federate Configuration Files
+
+The `encrypted` and `encryption_config` options can also be set from a federate configuration file, such as this:
+
+```json
+{
+  "encrypted": true,
+  "encryption_config": "/home/username/example-directory/openssl.json"
+}
+```
+
+### Federate Code
+
+Finally, it is also possible to set these options in the code of a federate (example shown in C++, but similar
+mechanisms can also be used from other language interfaces).
+
+Using a `FederateInfo` object named `fi`, this might look like:
+
+```cpp
+fi.encrypted = true;
+fi.encryptedConfig = "/home/username/example-directory/openssl.json";
+```
+
+Or, the federate info args string can provide the settings in the form of a string matching the command-line arguments
+described earlier:
+
+```cpp
+const std::string fedArgs =
+    "--core_type=tcp --encrypted --encryption_config=" + std::string(TEST_BIN_DIR) +
+    "encryption_config/openssl.json";
+helics::FederateInfo fi_enc(fedArgs);
+helics::ValueFederate fed1("fed1", fi_enc);
+```
+
+To check if the HELICS library being linked to has encryption support, the `helics-config.h` header can be imported,
+which will define `HELICS_ENABLE_ENCRYPTION` if encryption is supported:
+
+```cpp
+#include "helics/helics-config.h"
+
+#ifdef HELICS_ENABLE_ENCRYPTION
+// <code that requires HELICS compiled with encryption support>
+#endif
+```
+
+If CMake is being used, the following snippet (taken from the main HELICS repository `tests/helics/CMakeLists.txt`
+file around lines 37-46) shows how CMake's `configure_file` function can be used to automatically fill in the
+location of certificate and private key files (WARNING: do NOT put private key files in a public repository; ensuring
+you are using the encryption features securely is your own responsibility):
+
+```cmake
+if(HELICS_ENABLE_ENCRYPTION)
+    configure_file(
+        "test_files/encryption_config/openssl.json.in" "test_files/encryption_config/openssl.json"
+    )
+    configure_file(
+        "test_files/encryption_config/multiBroker_encrypted_bridge.json.in"
+        "test_files/encryption_config/multiBroker_encrypted_bridge.json"
+    )
+endif()
+```
+
+And the corresponding input `openssl.json.in` file looks like:
+
+```json
+{
+  "encrypted": true,
+  "verify_file": "${CMAKE_CURRENT_SOURCE_DIR}/test_files/encryption_config/openssl_certs/ca.pem",
+  "certificate_chain_file": "${CMAKE_CURRENT_SOURCE_DIR}/test_files/encryption_config/openssl_certs/server.pem",
+  "private_key_file": "${CMAKE_CURRENT_SOURCE_DIR}/test_files/encryption_config/openssl_certs/server.pem",
+  "tmp_dh_file": "${CMAKE_CURRENT_SOURCE_DIR}/test_files/encryption_config/openssl_certs/dh4096.pem",
+  "password": "test"
+}
+```
+
+Another big point of CAUTION: do NOT use the example/test SSL certificates or private key files provided in the HELICS
+repository! By doing so, you might as well not be using any encryption at all.
```

Comment:
Should add a note about why they are there then.  Obviously for testing purposes but we should add a note to that effect.

Reply:
I added a line to this saying they are for testing only.

Reply:
Maybe it would be good to just stick a warning as the first line of this page saying that if improperly configured, communication will not be secure, and in that same place caution against using any example/test certificate and private key files they find online/in the HELICS repository for real world use cases.